### PR TITLE
Bug 1441984 - Only show OpenInPrivateBrowsing option if private browsing enabled

### PR DESCRIPTION
**Pull request by k88hudson**
This should fix the issue of only showing OpenInPrivateBrowsing when private browsing is enabled, I decided to add the value into `Prefs` since `SnippetsFeed` could be turned off. Alternatively, I could add it to a completely new file/reducer, but i don't think overhead is worth it

cc @Mardak 

@rlr what do you think about connecting the `LinkComponent` again? I had to refactor tests to get it work

### CODE REVIEW

File: system-addon/lib/PrefsFeed.jsm [hunk 1]
```diff
@@ -63,6 +66,9 @@ this.PrefsFeed = class PrefsFeed {
       values[name] = this._prefs.get(name);
     }
 
+    // Not a pref, but we need this to determine whether to show private-browsing-related stuff
+    values.isPrivateBrowsingEnabled = PrivateBrowsingUtils.enabled;
```
Comment (k88hudson) [inline system-addon/lib/PrefsFeed.jsm]: @mkaply does this look right to you? Also, how would we go about watching for changes to this?

File: system-addon/content-src/lib/link-menu-options.js [hunk 2]
```diff
@@ -1,5 +1,15 @@
 import {actionCreators as ac, actionTypes as at} from "common/Actions.jsm";
 
+const _OpenInPrivateWindow = site => ({
+  id: "menu_action_open_private_window",
+  icon: "new-window-private",
+  action: ac.AlsoToMain({
```
Comment (rlr) [inline system-addon/content-src/lib/link-menu-options.js]: you didn't change this, but should it be `ac.OnlyToMain(...` instead?

File: system-addon/lib/PrefsFeed.jsm (hunk 1 above)
Comment (k88hudson) [reply system-addon/lib/PrefsFeed.jsm]: Ok perfect, thanks!

**Comment by as-pine-proxy** [comment]
[Treeherder: 32a0da06f91dec8ea7640b98b0706bdadfb60971](https://treeherder.mozilla.org/#/jobs?repo=pine&revision=32a0da06f91dec8ea7640b98b0706bdadfb60971)

**Pull request merged**